index.js

```diff
@@ -96,6 +107,50 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
+  if( typeof name === 'object') {
+    var cookies = name;
+    var serializeOptions = val;
+
+    var cookieNames = Object.keys(cookies);
+    if(0 === cookieNames.length) {
+      return '';
+    } else if(cookieNames.length > 1) {
+      // If there are more than one cookies to serialize, only allow
+      // an encoding option to be set
+      var opt = serializeOptions || {};
+      serializeOptions = {
+        encode: opt.encode || encode
+      };
+    }
+
+    var serializedCookies = [];
+    cookieNames.forEach(function(cookieName) {
+        serializedCookies.push(serializeNameValue(cookieName, cookies[cookieName], serializeOptions));
```

Comment:
We already know the number of cookies ahead time time; forcing the runtime to keep expanding the array is not very efficient.